F/faust/build_tarballs.jl

```diff
@@ -0,0 +1,193 @@
+
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "faust"
+version = v"2.32.16"
+
+# Collection of sources required to build faust
+sources = [
+    GitSource("https://github.com/grame-cncm/faust.git",
+              "53055b47f0ac81bee92b32db3c12729759e005f4"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/faust
+
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/disable_interp.patch
+CMAKE_FLAGS=()
+CMAKE_TARGET=${target}
+
+if [[ "${target}" == *musl* ]]; then
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/execinfo.patch
+    export CXXFLAGS="-DALPINE"
+fi
+
+if [[ "${target}" == *apple* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.9
+    export LDFLAGS="${LDFLAGS} -mmacosx-version-min=10.9"
+    # If we're building for Apple, CMake gets confused with `aarch64-apple-darwin` and instead prefers
+    # `arm64-apple-darwin`.  If this issue persists, we may have to change our triplet printing.
+    if [[ "${target}" == aarch64* ]]; then
+        CMAKE_TARGET=arm64-${target#*-}
+    fi
+fi
+
+if [[ "${target}" == *mingw* ]]; then
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/ws2.patch
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/windows-microhttpd.patch
+    export LDFLAGS="${LDFLAGS} -lws2_32 -lmicrohttpd"
+fi
+
+if [[ "${target}" == *freebsd* ]]; then
+    export LDFLAGS="${LDFLAGS} -lexecinfo"
+fi
+
+CMAKE_FLAGS+=(-DCMAKE_C_COMPILER_TARGET=${CMAKE_TARGET})
+CMAKE_FLAGS+=(-DCMAKE_CXX_COMPILER_TARGET=${CMAKE_TARGET})
+
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+
+if [[ "${target}" == "x86_64-linux-gnu" || "${target}" == "i686-linux-gnu" ]]; then
+    # Use llvm-config binary directly.
+    export PATH="$prefix/tools:$PATH"
+else
+    CMAKE_FLAGS+=(-DUSE_LLVM_CONFIG=OFF)
+    CMAKE_FLAGS+=(-DLLVM_DIR=${prefix}/lib/cmake/llvm)
+
+    if [[ "${bb_full_target}" == x86_64-linux-musl-*-cxx03 ]]; then
+        # For some reason this target requires "-lLLVM-11jl"
+        # while others require "-lLLVM" to build.
+        atomic_patch -p1 ${WORKSPACE}/srcdir/patches/set_llvm_libs_musl_cxx03.patch
+    else
+        atomic_patch -p1 ${WORKSPACE}/srcdir/patches/set_llvm_libs.patch
+    fi
+fi
```

Comment:
Wow, you did a massive work in this pull request!  If you feel like experimenting a bit more, you can try to add `LLVM` as a [`HostBuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.HostBuildDependency) and try to run `llvm-config` from there.  Being able to directly use `llvm-config` was actually the [original motivation](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/37) for introducing `HostBuildDependency`.